libs/ledger/src/upow/synergetic_execution_manager.cpp

```diff
@@ -158,14 +163,24 @@ bool SynergeticExecutionManager::ValidateWorkAndUpdateState(std::size_t num_lane
   return true;
 }
 
-void SynergeticExecutionManager::ExecuteItem(WorkQueue &queue, ProblemData const &problem_data,
+void SynergeticExecutionManager::ExecuteItem(WorkQueue &queue, ProblemData &problem_data,
                                              std::size_t num_lanes)
 {
   ExecutorPtr executor;
 
   // pick up an executor from the stack
   {
     FETCH_LOCK(lock_);
+    if (executors_.empty())
+    {
+      FETCH_LOG_WARN(LOGGING_NAME, "Executors empty, can't execute item!");
+      auto ptr          = std::make_shared<WorkItem>();
+      ptr->problem_data = std::move(problem_data);
+      ptr->work_queue   = std::move(queue);
+      solution_stack_.emplace_back(std::move(ptr));
+      no_executor_count_->increment();
+      return;
+    }
```

Comment:
I suspect this is just a racy thing. I would not add the element to the solution stack. I would sit this thread in a loop for a short period of time. If this fails then I would fail hard.

Reply:
Changed the logic